packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -71,20 +88,31 @@ export class Span implements api.Span, ReadableSpan {
     parentTracer: Tracer,
     context: Context,
     spanName: string,
-    spanContext: api.SpanContext,
-    kind: api.SpanKind,
+    spanContext: SpanContext,
+    kind: SpanKind,
     parentSpanId?: string,
-    links: api.Link[] = [],
-    startTime?: api.TimeInput,
-    clock: Clock = otperformance,
+    links: Link[] = [],
+    startTime?: TimeInput,
+    _deprecatedClock?: unknown, // keeping this argument even though it is unused to ensure backwards compatibility
   ) {
-    this._clock = clock;
     this.name = spanName;
     this._spanContext = spanContext;
     this.parentSpanId = parentSpanId;
     this.kind = kind;
     this.links = links;
-    this.startTime = timeInputToHrTime(startTime ?? clock.now());
+
+    const now = Date.now();
+    this._performanceStartTime = otperformance.now();
+    this._performanceOffset = now - (this._performanceStartTime + getTimeOrigin());
+
+    // if startTime is a number smaller than the start of the process
+    // assume it is a performance API timestamp and apply correction as needed
+    if (typeof startTime === 'number' && startTime < getTimeOrigin()) {
```

Comment:
What if `startTime` is an HrTime?  IIUC it should be compared to the time origin too.

Reply:
are there any browser performance APIs which return a HrTime timestamp? That's the only time we need to apply a correction.

Reply:
PTAL here https://github.com/open-telemetry/opentelemetry-js/issues/3279#issuecomment-1320064182